### PR TITLE
05584 - automating address book initialization tests

### LINKED ISSUE #5584: Create automated integration test for address book initialization
Automated Testing (mirror manual testing)  
- [ ] Whole network from genesis, validate use of config address book.  
- [ ] node restart, no upgrade, validate saved sign state used. 
- [ ] node restart on upgrade boundary, validate swirld state initialization of address book.
- [ ] config flag for forcing address book
- [ ] testing node with 0 stake updated behavior. 

Verification Steps
- [ ] swirlds.log output. 
- [ ] network behavior changes with different stake weights on nodes. 
- [ ] (optional) address books written to disk. 

Single Test Run ?  (As many as needed) 
1. genesis.
1. restart with existing state. 
1. restart with upgrade boundary. 

Avoid writing new test runners if possible. 
Test by hand initially before JRS testing. 
Software versions are coming from the configurations in settings.txt.

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
**Description**:
The following changes allow automated testing through JRS without recompiling.
* moves `isOwnHost` check from `equalsWithoutStake` to `equals` in Address.java
* adds AddressBookTestingToolConfig.java with `softwareVersion` and `stakingBehavior` settings.
* updates README.md instructions to use the new settings instead of recompiling.
* updates softwareVersion through settings in AddressBookTestingToolMain.java
* updates stakingBehavior through settings in AddressBookTestingToolState.java
* validates the different testing scenarios in AddressBookTestingToolState.java

**Related issue(s)**:

Fixes #5584 

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java [hunk 1]
```diff
@@ -57,9 +76,17 @@
 public class AddressBookTestingToolState extends PartialMerkleLeaf implements SwirldState, MerkleLeaf {
 
     private static final Logger logger = LogManager.getLogger(AddressBookTestingToolState.class);
+    //
+    //    /** modify this value to change how updateStake behaves. */
+    //    private final int stakingBehavior;
+    //
+    //    /** modify this value to change the test scenario being run and validated. */
+    //    private final int testScenario;
```
Comment (cody-littley) [inline platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java:84]: remove

**Review by cody-littley — APPROVED** [review]

**Review by alittley — APPROVED** [review]

File: platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java (hunk 1 above)
Comment (edward-swirldslabs) [reply platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java:84]: removed

File: platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/Address.java [hunk 2]
```diff
@@ -831,7 +831,6 @@ public boolean equals(Object o) {
      */
```
[the remark this answers is not in the document]
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-common/src/main/java/com/swirlds/common/system/address/Address.java:831]: renamed method and updated comments.

**Review by alittley — DISMISSED** [review]

**Review by timo0 — APPROVED** [review]

**Review by timo0 — APPROVED** [review]

File: platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java [hunk 3]
```diff
@@ -58,8 +77,13 @@ public class AddressBookTestingToolState extends PartialMerkleLeaf implements Sw
...
 
-    /** modify this value to change how updateStake behaves. */
-    private static final int STAKING_PROFILE = 1;
+    /** the suffix for the debug address book */
+    private static String DEBUG = "debug";
+
+    private final AddressBookTestingToolConfig testingToolConfig =
+            ConfigurationHolder.getConfigData(AddressBookTestingToolConfig.class);
+    /** the address book configuration */
+    private final AddressBookConfig addressBookConfig = ConfigurationHolder.getConfigData(AddressBookConfig.class);
```
Comment (cody-littley) [inline platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java:86]: Is it possible to do this without the ConfigurationHolder? We are actively trying to deprecate static access to configuraiton.

It should be possible to get the configuration inside the init() method, if that doesn't work then let me know and we can brainstorm ways to improve things.

Comment (edward-swirldslabs) [reply platform-sdk/platform-apps/tests/AddressBookTestingTool/src/main/java/com/swirlds/demo/addressbook/AddressBookTestingToolState.java]: the config is now retrieved through init.

**Review by timo0 — APPROVED** [review]

**Pull request merged**